### ISSUE TITLE
Not Working 

### ISSUE THREAD

**Issue by kadevland** [Dickurt/vscode-php-formatter#9]
I have PHP-CS-Fixe installed by composer globaly, in terminal i have access to php-cs-fix 
i have configurate key bind for art-shitf-f, but nothing append juste VSC see file modificated.
But nothing fixed, is the same file.
If phpformatter.useTempFiles set false, nothing append too  only close file.

**Comment by rgfx** [comment]
I dont have composer, added php-cs-fixer to my xampp dir, and I have the same issues aswell. Windows 7

"phpformatter.pharPath": "C:/xampp/php/php-cs-fixer",

**Comment by kadevland** [comment]
I m under debian

**Comment by kadevland** [comment]
I trie path tu php-cs-fixer and no working too

**Comment by rgfx** [comment]
Do I have to install php-cs-fixer if am not using composer? Or just point to file. 

user

    "phpformatter.phpPath": "C:/xampp/php/php.exe",

    "phpformatter.pharPath": "C:/xampp/php/php-cs-fixer",

keybinding

[{
    "key": "alt+shift+f",
    "command": "phpformatter.fix",
    "when": "editorFocus"
}]

Am I missing something, the command phpformatter.fix is not present when I hit F1.

I would really like this to work, any assistance would be greatly appreciated.

**Comment by Dickurt** [comment]
Would you mind posting all your settings regarding `phpformatter`? Also, you said the php-cs-fixer command is available from the command line, does that mean from anywhere, i.e. was it added to your PATH variable?

I think the problem is that you forgot to set the `phpformatter.composer` setting to `true`.

**Comment by rgfx** [comment]
I am not using composer. Upon further inspection I don't think php-cs-fixer is working correctly for me.  When I do things manually I get "Fixed all files in 0.087 seconds, 6.000 MB memory used"

But the files do  change :(

**Comment by Dickurt** [comment]
@rgfx What you're saying is contradictory. You say "I don't think php-cs-fixer is working correctly" and then "the files do change." So I'll assume that PHP-CS-Fixer is correctly installed.

Could you share the file that you are trying to format (preferably the before _and_ after version). Check out the [Troubleshooting](https://github.com/Dickurt/vscode-php-formatter/wiki/Troubleshooting#my-code-wont-format-does-the-extension-even-work) section on the [wiki](https://github.com/Dickurt/vscode-php-formatter/wiki). There might be something you could use.

**Comment by rgfx** [comment]
The manual command looked like it worked, but it never changed the file. I don't think it an issue with your extension, rather php-cs-fixer.

However  'phpformatter.fix 'never showed up in the command pallet.

  I purposely messed up the PHP file nothing changed. 

				<?php 
				
				endwhile;
    endif;

		wp_reset_query(); ?>

I really appreciate you trying to help me.

**Comment by Dickurt** [comment]
I see. That's indeed something outside of the control of this plugin. Try looking at the [PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer#usage) fixer options, perhaps there is something that provides formatting for code like this.

**Comment by Dickurt** [comment]
@kadevland Are you still running into this issue?

**Comment by rgfx** [comment]
I looked at the docs again, nothing new for me there.  Someone with a similar issue got it working with composer. 

 I've spent far to much time on PHP formatting as it, going to let it rest for a bit.  come back to it after I finish my current project. 

Feel free to close this issue. I'll reopen when I get a chance to dig deeper. Its not even my thread, probably starting to annoy kadevland.

Thanks for checking in.

**Comment by kadevland** [comment]
@Dickurt  i will retesting with composer and said you after

**Comment by Dickurt** [comment]
@rgfx No problem at all! Good luck with your current project.

**Comment by Dickurt** [comment]
@kadevland Any success yet?

**Comment by Dickurt** [comment]
Closing this because `phpformatter.useTempFiles` is deprecated since the latest version. In the same version, issues like these have been added to the wiki [Troubleshooting ](https://github.com/Dickurt/vscode-php-formatter/wiki/Troubleshooting) section.

**Comment by jon-heller** [comment]
I realize this issue is closed, but I'm having the exact same problem.

At first, when I installed php-cs-fixer using composer, nothing at all would happen when I triggered this extension in VSCode. I instead switched to using a .phar file, and the file shows as modified in VS Code but nothing changes.

Running php-cs-fixer from the command line works as expected. 

Logging is enabled, but nothing appears in the VS Code console.

**Comment by Dickurt** [comment]
@jon-heller That's strange. Are you using a keyboard shortcut to trigger the formatter, as well? And by console you mean VSCode's developer console, right?

The extension should output something. The fact that it doesn't means it never runs at all. So I'm wondering why it marks the file as dirty. This will be a tough one!

**Comment by jon-heller** [comment]
@Dickurt Thanks for the quick response! Yes, I'm using a keyboard shortcut. I don't see any commands related to php formatter in the command pane -- is that maybe a symptom of the problem as well?.

**Comment by Dickurt** [comment]
@jon-heller By "command pane", do you mean the contributions tab with all of the extension's settings? Or the other tab with "commands"? The tab with commands should normally have one command called "fix". If that is not present. Then we're on to something.

**Comment by jon-heller** [comment]
@Dickurt Ah sorry using some Sublime Text terminology -- I meant command palette (shift-option P). 

But now that I see the command is "fix", I do see it listed there ("Fixes current file or selection if there is any"). Same issue though, of it "touching" the file but not making any changes.

**Comment by Dickurt** [comment]
@jon-heller Ah, I see now. This is a mystery to me. Would it be possible for you to take a screenshot of the file you are fixing? And the console output in VSCode? Additionally, could you show me the output when you fix the file using the command line, like you said? Also, what are your VSCode user settings, what is your OS, version of VSCode, version of PHPFormatter, version of PHP CS Fixer? Most importantly, what is the color of your socks.

**Comment by jon-heller** [comment]
@Dickurt So, most importantly: My socks today are tan. Sometimes they are closer to brown, but today they're definitely tan. I think it was a wise choice.

The file I'm trying to fix is actually in the example you gave [here](https://github.com/Dickurt/vscode-php-formatter/wiki/Troubleshooting), in the interest of keeping things simple. The output I get after running php-cs-fixer from the command line gives the same result in your example. 

User settings:

```
{
    "php.validate.executablePath": "/usr/local/bin/php",
    "phpcs.standard": "PSR1",
    "beautify.language": {
      "html": ["twig", "htm", "html"]
    },
    "editor.scrollBeyondLastLine": false,
    "phpformatter.composer": false,
    "phpformatter.logging": true,
    "phpformatter.pharPath": "/Users/Jon/Downloads/php-cs-fixer"
}
```

OS is Mac Sierra 10.12.1, VS Code 1.7.2, PHP Formatter 0.2.3, and PHP CS Fixer 2.0.0

**Comment by Dickurt** [comment]
@jon-heller Definitely an interesting choice of socks. Very much in line with the weather around here. I'm sporting pitch blacks with a fine, pure grey line.

As for the error at hand here, it could be all sorts of things. For instance, VSCode has an API for registering your extension as formatter. That API chooses which formatter is more suited to formatting the file at hand. So that if there are several formatters that could potentially fix a file, one gets precedence over the others. So even though Beautify is configured to format the given filetypes only, it could still take precedence over the other formatters installed by accident.

*However*, having said that, I don't feel comfortable blaming other extensions for an issue that is likely mine. My final assumption, therefore, is that there are some issues with the new version of PHP CS Fixer that I haven't encountered yet. Which has only very recently been released. So I haven't come around to testing it yet.

I hope to look into this soon enough! Thank you for your detailed report. That definitely helps me out.

**Comment by jon-heller** [comment]
@Dickurt Well if I can be of any more help with testing, let me know -- really appreciate your work on this extension.

Also for what it's worth I did try disabling all my other extensions with the thought that maybe there was some kind of conflict. No luck though.

**Comment by yaliv** [comment]
I'm running Zorin OS (Ubuntu based).

Able to invoke `php-cs-fixer fix myFile.php` via terminal and it works well.

But not working with VS Code. I've installed the plugin and configured a keybinding. After pressing the button, no change in my source code, but the file turned into modified state.

**Comment by Dickurt** [comment]
@jon-heller Thank you, I appreciate you saying that. This extension is a one-man effort. So it's rough at times to see all the issues crop up.

@yaliv Are you using PHP CS Fixer 2.0.x as well? I suspect that to be the issue. I will try to work something out soon enough.

**Comment by vysker** [comment]
I just released version 0.2.4. Let me know if that solves the issue for you guys. I have yet to look at PHP CS Fixer 2.0.0, though. That was not the focus of this release.

**Comment by yaliv** [comment]
@Dickurt Oh yes, I use PHP CS Fixer 2.0.0.

I've updated the plugin to version 0.2.4, still the same behavior, whether using composer or using `php-cs-fixer.phar` file.

**Comment by rgfx** [comment]
Wow this thread got big.  Your updates and php-fixer updates inspired me to give it another whirl. I still have the same issue.

This screen shoot should give you good idea of my setup, and issue. 

![php fixer issue](https://i.imgsafe.org/e329183a9c.jpg)

**Comment by vysker** [comment]
@rgfx Thank you for taking the time to post the screenshots. That's really helpful.

Now I can finally see what the issue is with the command palette! If you type `fix` in the command palette, as opposed to `php`, the command _will_ show up. You see, the command is called `phpformatter.fix`, and apparently the command palette only matches against the part _after_ the extension name, i.e. `fix`. I assumed it would match against `phpformatter` as well. Unfortunately it doesn't.

**TL;DR: Type `fix` instead of `php` in the command palette.**

**Comment by Join-Git-Hub** [comment]
Hello

I wanted to try php formater (Windows 7) but it just does not work.

The code is unfortunately not formatted, it changes nothing!

Can someone help me please?

Thank you so much!
![php](https://cloud.githubusercontent.com/assets/17765696/21183070/783291f4-c206-11e6-92d5-1b355aff3e03.jpg)
![php_1](https://cloud.githubusercontent.com/assets/17765696/21183071/7837dc72-c206-11e6-9803-83843aac273f.jpg)

**Comment by vysker** [comment]
@Join-Git-Hub Hi, thank you for reporting in. Could you tell me your PHP CS Fixer version? Thanks!

**Comment by kadevland** [comment]
i will retry this week-end, sorry i was lot busy :)

**Comment by Join-Git-Hub** [comment]
@Dickurt and @kadevland - Thanks for the quick response! 
I have download the PHP CS Fixer Version from here:
[https://github.com/FriendsOfPHP/PHP-CS-Fixer#installation](url)
It seem this is the Version 2.0.0.

**Comment by vysker** [comment]
@Join-Git-Hub Alright, thanks for the info!

To everyone: I think there is a clear pattern here. Version 2.0.x of PHP CS Fixer is causing a lot of trouble. So I will open up a new issue to investigate this further.

**Edit:** Please continue the discussion over at [the new thread](https://github.com/Dickurt/vscode-php-formatter/issues/15).
